sale_triple_discount/models/sale_order_line.py

```diff
@@ -0,0 +1,158 @@
+        store=True,
+        readonly=False,
+        string="Disc. 2 (%)",
+        digits="Discount",
+        default=0.0,
+    )
+    discount3 = fields.Float(
+        compute="_compute_discounts",
+        precompute=True,
+        store=True,
+        readonly=False,
+        string="Disc. 3 (%)",
+        digits="Discount",
+        default=0.0,
+    )
+    discounting_type = fields.Selection(
+        selection=[("additive", "Additive"), ("multiplicative", "Multiplicative")],
+        default="multiplicative",
+        required=True,
+        help="Specifies whether discounts should be additive "
+        "or multiplicative.\nAdditive discounts are summed first and "
+        "then applied.\nMultiplicative discounts are applied sequentially.\n"
+        "Multiplicative discounts are default",
+    )
+
+    @api.constrains("discounting_type")
+    def _discounting_type_additive_not_allowed(self):
+        # FIXME: see https://github.com/OCA/sale-workflow/issues/3649
+        if any(rec.discounting_type == "additive" for rec in self):
+            raise ValidationError(
+                _(
+                    "Additive discount type is not fully implemented."
+                    " See https://github.com/OCA/sale-workflow/issues/3649 "
+                )
+            )
+
+    def _get_final_discount(self):
+        self.ensure_one()
+        if self.discounting_type == "additive":
+            return self._additive_discount()
+        elif self.discounting_type == "multiplicative":
+            return self._multiplicative_discount()
+        else:
+            raise ValidationError(
+                _("Sale order line %(name)s has unknown discounting type %(dic_type)s")
+                % {"name": self.name, "disc_type": self.discounting_type}
+            )
+
+    def _additive_discount(self):
+        self.ensure_one()
+        discount = sum(self[x] or 0.0 for x in self._discount_fields())
+        if discount <= 0:
+            return 0
+        elif discount >= 100:
+            return 100
+        return discount
+
+    def _multiplicative_discount(self):
+        self.ensure_one()
+        discounts = [1 - (self[x] or 0.0) / 100 for x in self._discount_fields()]
+        final_discount = 1
+        for discount in discounts:
+            final_discount *= discount
+        return 100 - final_discount * 100
+
+    @api.model
+    def _discount_fields(self):
+        return ["discount1", "discount2", "discount3"]
+
+    @api.depends("discount1", "discount2", "discount3", "discounting_type")
+    def _compute_discount(self):
+        res = super()._compute_discount()
+        for rec in self:
+            rec.discount = rec._get_final_discount()
+        return res
+
+    def _inverse_discount(self):
+        for rec in self:
+            rec.update({"discount1": rec.discount, "discount2": 0, "discount3": 0})
+
+    @api.depends("discount")
+    def _compute_discounts(self):
+        super()._compute_discount()
```

Comment:
You're calling a super for a different method, is this intended?

Reply:
yes, I will add a comment to explain why, when the discount is changed directly or via its compute method, it triggers the compute methods for the three discount fields. At this point, we need to make sure the cache is coherent before calling the inverse. calling _compute_discount() directly would alter the discount value by recomputing it from the three discounts, which may still be outdated, calling super() on the other hand avoids that